python-avd/pyavd/_eos_designs/shared_utils/routing.py

```diff
@@ -99,6 +99,29 @@ def isis_instance_name(self: SharedUtilsProtocol) -> str | None:
         # This point cannot be reached because the function won't be called if either of the conditions in the if-block is not satisfied.
         return None
 
+    @cached_property
+    def bgp_as_notation(self: SharedUtilsProtocol) -> Literal["asdot", "asplain"]:
+        bgp_as_notation = self.inputs.bgp_as_notation
+        if bgp_as_notation == "asdot" or (bgp_as_notation == "auto" and "." in str(self.bgp_as)):
+            return "asdot"
+        return "asplain"
+
+    def get_asn(self: SharedUtilsProtocol, asn: str | int | None) -> str | None:
+        if asn is None:
+            return None
+
+        bgp_as_notation = self.bgp_as_notation
+        if bgp_as_notation == "asdot" and "." not in str(asn):
```

Comment:
Can this fail if I input `bgp_as: blahblha` (not dot, a string) what is the error then?

Reply:
Done!